packages/metal-dom/test/dom.js

```diff
@@ -688,11 +755,30 @@ describe('dom', function() {
 				dom.triggerEvent(input, 'click');
 				dom.triggerEvent(select, 'click');
 				dom.triggerEvent(button, 'click');
+				dom.triggerEvent(textarea, 'click');
 				assert.strictEqual(0, listener1.callCount);
 				assert.strictEqual(0, listener2.callCount);
 				assert.strictEqual(0, listener3.callCount);
 				assert.strictEqual(0, listener4.callCount);
 			});
+
+			it('should not run listeners of the click event to an element which its valid parent is disabled using "dispatchEvent()"', function() {
+				var element = document.createElement('div');
+				element.innerHTML = '<button class="nomatch" disabled>' +
+					'<span class="match"></span></button>';
+				document.body.appendChild(element);
+
+				var listener = sinon.stub();
+				dom.delegate(element, 'click', '.match', listener);
+
+				var eventObj = document.createEvent('HTMLEvents');
+        		eventObj.initEvent('click', true, true);
+
+        		var target = element.querySelector('.match');
+        		target.dispatchEvent(eventObj);
+
+				assert.strictEqual(0, listener.callCount);
```

Comment:
This test just made me notice something weird. What this test is doing is:

1. Creating a disabled button with a span content.
2. Listening to clicks on the span.
3. Checking that the listener isn't called.

The problem is that in the browser, that listener is supposed to be called. I just tried this same use case myself in Chrome, and clicking the **span** will still trigger the span's listener, it will only stop triggering the **button's** listener, since that's the element that was disabled.

So the fact that this test is passing shows that something is wrong with the logic. The logic should be such that the **span** will still trigger listeners, but not the disabled **button**. There should be a test that listens to delegate on the **button** (`dom.delegate(element, 'click', 'button', listener)` for example), and makes sure that clicking the **span** does not cause that listener to be triggered. If the listener is on the **span** it should keep being triggered though, so this test should be kept, but also updated to check that the `callCount` is 1, not 0.

This seems to be showing that `isAbleToInteractWith_` is doing the wrong check. The [link]( https://html.spec.whatwg.org/#concept-fe-disabled) that you posted on the issue explaining how this should work seems to also be saying this. It basically says that a disabled element should prevent clicks on itself, and that disabled elements are only **button, input, select or textarea** that are disabled (not their children too). It also mentions descendants of disabled **fieldset** but I also tested this on Chrome, and could only reproduce when these descendants are also **button, input, select or textarea**, the others still get the event triggered regardless of the parent.

So I think that we need to change  `isAbleToInteractWith_` to only return false if the given node is **button, input, select or textarea**, and is either disabled or a child of a disabled fieldset. Then, in `handleDelegateEvent_` this should be called for each listener triggered in each element, not just for the first target, so probably inside `triggerListeners_` instead of `handleDelegateEvent_`.

Let me know if you have any questions about this, it's hard to explain it all in a single comment :)